src/lib/helpers/uploader.spec.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * @jest-environment ./test/custom.testenv.js
+ *
+ * Necessary due to JSDOM not providing TextDecoder
+ * https://stackoverflow.com/a/57713960
+ */
+import imageCompression from 'browser-image-compression';
+import { create } from 'ipfs-http-client';
+
+import { imageUpload, ipfsUpload } from './uploader';
+
+jest.mock('ipfs-http-client');
+jest.mock('browser-image-compression');
+
+const file = new File(['FileContents'], 'file-name');
+describe('#ipfsUpload', () => {
+  it('should pin images', async () => {
+    const addMock = jest.fn();
+    // TODO: fix "does not exist on type... IDE error"
+    create.mockReturnValue({
+      add: addMock,
+    });
+
+    const upload = await ipfsUpload(file, {});
+
+    expect(addMock).toHaveBeenCalledWith(file, { pin: true });
+    expect(upload).toEqual(undefined);
+  });
+});
+
+describe('#imageUpload', () => {
+  it('should throw an error on invalid input', async () => {
+    await expect(
+      imageUpload(
+        new File(['sdfasdf'], 'file-name', {
+          type: 'zip',
+        }),
+        {
+          ipfsClientOptions: {},
+        }
+      )
+    ).rejects.toThrowError('File provided is of type "zip".');
+  });
+
+  it('should pin files when using IPFS', async () => {
+    const { file, addMock, drawFileInCanvasSpy } = mockDependencies();
+    await imageUpload(file, { ipfsClientOptions: {} });
+
+    expect(imageCompression).toHaveBeenCalledTimes(5);
```

Comment:
What is the link between "it should pin files when using IPFS" and "imageCompression is called 5 times" ?

I don't see any obvious link.

Also, the 5 comes from the fact that there are 5 file formats, but it may come as a more obvious thing in the code because some people may wonder "why 5?".

Also, adding a new image size somewhere else in the code will very likely break this test but it is not related.

I suggest to either:
- remove this check from this test
- to give a better name
- to add a comment which explains why it is 5
- use something like `.toHaveBeenCalledTimes(sizes.length);` or similar